src/components/Button.tsx

```diff
@@ -42,7 +46,8 @@ export function Button({
         className,
       )}
       disabled={disabled}
-      onClick={disabled ? undefined : onClick}
+      type={submit ? 'submit' : 'button'}
+      onClick={disabled || !onClick ? undefined : onClick}
```

Comment:
I don't think `|| !onClick` is necessary here

Reply:
Sure!